### ISSUE TITLE
DataGridView sends unnecessary event to screen readers on every cell change.

### ISSUE THREAD

**Issue by trypsynth** [dotnet/winforms#12867]
### .NET version

9.1.00

### Did it work in .NET Framework?

Not tested/verified

### Issue description

I have this simple Windows Forms app using DataGridView. Every time I press an arrow key to move to a different cell, using either NVDA or Narrator, after reading all of the correct information about the cell, the screen reader will say "edit", for seemingly no reason.
```
public class Form1 : Form {
	private DataGridView gridView;

	public Form1() {
		Text = "Periodic Table";
		ClientSize = new Size(1000, 600);
		gridView = new DataGridView {
			AllowUserToAddRows = false,
			AllowUserToDeleteRows = false,
			AllowUserToOrderColumns = false,
			AllowUserToResizeColumns = false,
			AllowUserToResizeRows = false,
			ColumnCount = 18,
			RowCount = 11,
			Dock = DockStyle.Fill,
			ReadOnly = true,
			RowHeadersVisible = true,
			ColumnHeadersVisible = true
		};
		Controls.Add(gridView);
	}

	[STAThread]
	public static void Main() {
		ApplicationConfiguration.Initialize();
		Application.Run(new Form1());
	}
}
```

### Steps to reproduce

1. Run the demo app from above with a screen reader active.
2. Press the arrow keys to move around the grid.

**Comment by Zheng-Li01** [comment]
@TheQuinbox , this is not a real issue. for now, the keyboard tooltip is supported for the controls by default, so the **Edit** tooltip will be announced when using the arrow keys got though the cells as below screenshot. So you can disable the tooltip such as below code section.
` ShowCellToolTips = false`

![Image](https://github.com/user-attachments/assets/a37c39f1-ec35-4401-8b88-ea8630574b0b)

**Comment by TheQuinbox** [comment]
Hi @Zheng-Li01,
You are indeed correct, adding this removed the unnecessary event being fired. What exactly is the purpose of it, though? As a screen reader user myself, I don't expect to hear "edit" when navigating around a grid that doesn't actually have any editable content in it.
Thanks!

**Comment by Zheng-Li01** [comment]
@TheQuinbox, thanks for your response. as below screenshot, sometimes the screen reader user wants to know which type of cell was used for now, who can get the cell's type info by using screen reader to announce the tooltip.
![Image](https://github.com/user-attachments/assets/faa4e37f-c8f5-4f01-b609-7fef56a3f737)
![Image](https://github.com/user-attachments/assets/852db2ef-16b4-4e29-8c28-6a5f539b83c9)

**Comment by TheQuinbox** [comment]
Hi @Zheng-Li01,
this is exactly my point. Sorry, I should've been clearer. My DataGridView is set to read-only. Why, then, is the tool tip "edit", even though the user is unable to edit any content, and furthermore, they can't even navigate the cells like edit fields because the arrow keys  move around the grid.
Thanks,
Quin.

**Comment by LeafShi1** [comment]
@TheQuinbox @Tanya-Solyanik  Referring to the display of the Textbox control, when the Textbox is read-only, adding the ReadOnly attribute to it, do you think this is reasonable?

![Image](https://github.com/user-attachments/assets/40c156ed-547b-4691-8b7f-b2952783f38a)

![Image](https://github.com/user-attachments/assets/2440dc0d-12f4-4bd5-a3cc-b255de78e0bb)

**Comment by LeafShi1** [comment]
The tooltips of all controls will be read out automatically, which is controlled by the [AnnounceText](https://github.com/dotnet/winforms/blob/12d120b51b52cdb0bf27de5cdd406f52e52dfd67/src/System.Windows.Forms/src/System/Windows/Forms/ToolTip/ToolTip.cs#L624) method of Tooltip itself

The reason why DataGridViewCell displays Tooltip is that the [GetDefaultToolTipText](https://github.com/dotnet/winforms/blob/12d120b51b52cdb0bf27de5cdd406f52e52dfd67/src/System.Windows.Forms/src/System/Windows/Forms/Controls/DataGridView/DataGridViewTextBoxCell.cs#L320) method is overridden in `DataGridViewTextBoxCell`

@Tanya-Solyanik @TheQuinbox There are currently two solutions below. Which one do you think is more reasonable?

**Solution1**: we can disable the Tooltip of `DataGridViewCell`, so that no extra text will be read out. 
The following is the pronunciation result of NVDA after disabling the ToolTip of DataGridViewTextBoxCell

![Image](https://github.com/user-attachments/assets/d4e096b1-a628-4fed-bc84-840af79cdb69)

**Solution 2**: when the Textbox is read-only, adding the ReadOnly attribute to it's ToolTip

![Image](https://github.com/user-attachments/assets/340cec5e-461c-4cef-8fe5-72a090a54b21)

**Comment by LeafShi1** [comment]
For other application, like Excel, there is no Tooltip when focusing on a cell

![Image](https://github.com/user-attachments/assets/01b7109b-638d-4767-9040-32108d61bc21)

and no Tooltip in .NetFramework project, but the statue of the cell will be announced

![Image](https://github.com/user-attachments/assets/1946e594-38cc-4073-8709-d791e955375a)

**Comment by TheQuinbox** [comment]
Hi,
I can't see the results you posted, I'm completely blind, but the first
solution sounds reasonable to me.
Thanks.

**Comment by merriemcgaw** [comment]
We agree. I don't think this is a violation of any WCAG requirements, and honestly I don't believe that ToolTips in a data cell make sense any longer now that WCAG2.2 requires that Tooltips appear in keyboard navigation (plus be persistent and dismissible). In the case of a DataGridView having ToolTips turned on for the cells leads to a degraded experience for both screen reader users and sighted users!
